### PR TITLE
Feature: iCloud Drive and Photos namespace detection

### LINKED ISSUE #10805: iCloud Photos in Cloud Drives
### What feature or improvement do you think would benefit Files?

A link to iCloud Photos should be added to Cloud Drives, since that is likely the main usecase for iCloud when the user uses a Windows PC.

### Requirements

- Show iCloud Photos in the Cloud Drives menu
- Probably rename iCloud to iCloud Drive for consistency
  - I would've posted this as a separate issue, but it's not really a bug or a feature request, and you seem to be incredibly strict about using the templates, sooo...

### Files Version

Version: 2.4.4.0

### Windows Version

Windows 11 Pro 22H2 22623.1037

### Comments

iCloud Photos seems to be located at `%UserProfile%\iCloudPhotos`, right next to iCloud Drive.

In Windows Explorer, iCloud provides links in the sidebar to both iCloud Drive and iCloud Photos.

### PR DESCRIPTION
**Pull request by FayneAldan**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Changes iCloud Drive to use namespace detection, so it can detect a moved iCloud Drive location.
- Changes "iCloud" to "iCloud Drive" for consistency. (Drive is a service within iCloud and not the entirety of the service)
- Closes #10805

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
image

### CODE REVIEW

File: src/Files.App/Helpers/CloudDrivesDetector.cs [hunk 1]
```diff
@@ -92,6 +92,8 @@ private static Task<IEnumerable<ICloudProvider>> DetectGenericCloudDrive()
 						"Amazon Drive" => CloudProviders.AmazonDrive,
 						"Nextcloud" => CloudProviders.Nextcloud,
 						"Jottacloud" => CloudProviders.Jottacloud,
+                        "iCloudDrive!S-1-5-21-4064190285-3015957005-1528788272-1001!Personal" => CloudProviders.AppleCloudDrive,
```
Comment (gave92) [inline src/Files.App/Helpers/CloudDrivesDetector.cs:95]: Mm this looks a lot like a user SID. Are you sure is the same for everyone? To check: what does this cmd command print? `whoami /user`.

Comment (FayneAldan) [reply src/Files.App/Helpers/CloudDrivesDetector.cs:95]:
```
❯ whoami /user

USER INFORMATION
----------------                                                                                                        
User Name      SID
============== ==============================================
ethan-pc\ethan S-1-5-21-4064190285-3015957005-1528788272-1001
```
i hate everything.

File: src/Files.App/Helpers/CloudDrivesDetector.cs [hunk 2]
```diff
@@ -78,6 +78,12 @@ private static Task<IEnumerable<ICloudProvider>> DetectGenericCloudDrive()
 						driveType = appName;
 					}
 
+					// iCloud specific
```
Comment (yaira2) [inline src/Files.App/Helpers/CloudDrivesDetector.cs:81]: Shouldn't this go in the AppleCloud detector file?

Comment (FayneAldan) [reply src/Files.App/Helpers/CloudDrivesDetector.cs:81]: There's already detection in place for namespace, which is why I removed AppleCloudDetector. If you're referring to the fact that I added code specifically for iCloud, then shouldn't Nextcloud (right above) also have its own detector class?

Comment (yaira2) [reply src/Files.App/Helpers/CloudDrivesDetector.cs:81]: @gave92

**Review by gave92 — APPROVED** [review]

File: src/Files.App/Helpers/CloudDrivesDetector.cs (hunk 2 above)
Comment (gave92) [reply src/Files.App/Helpers/CloudDrivesDetector.cs:81]: It's ok, cloud drives that can be detected from `Explorer\Desktop\NameSpace` registry should go in DetectGenericCloudDrive()

**Comment by yaira2** [comment]
LGTM

**Pull request merged**